crons/image-monitor.sh

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env bash
+#
+# Find if a new tag has been published to dockerhub after the given tag.
+# Only works for new tags, not tag updates.
+# For tag updates, use Diun
+#
+set -eufo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "$0")" && pwd)"
+
+ERROR_CODE_INVALID_USAGE=1
+ERROR_CODE_INVALID_INPUT=2
+
+SLACK_URL="https://slackbot.internal.aleemhaji.com/message"
+SLACK_CHANNEL="CKE1AKEAV"
+
+ignore_tags="latest edge nightly beta preview unstable dev stable testing"
+global_ignore_tags_selector=""
+for rtag in $ignore_tags; do
+    global_ignore_tags_selector="$global_ignore_tags_selector | select(test(\"^$rtag$\") | not)"
+done
+
+all_ignore_tags="$(jq -r '. | tostring' "$SCRIPT_DIR/image-monitor-ignore.json")"
+
+slack() {
+	curl -sS -X POST -H "X-SLACK-CHANNEL-ID: ${SLACK_CHANNEL}" -d "$@" "$SLACK_URL"
+}
+
+function check_repository() {
+    if [ $# -ne 1 ]; then
+        echo >&2 "Usage:"
+        echo >&2 "  $0 <tag>"
+        return $ERROR_CODE_INVALID_USAGE
+    fi
+
+    registry_repository_tag="$1"
+    registry_repository="$(awk -F: '{print $1}' <<< "$registry_repository_tag")"
+    tag="$(awk -F: '{print $2}' <<< "$registry_repository_tag")"
+
+    possible_registry="$(awk -F/ '{print $1}' <<< "$registry_repository")"
+    if nslookup "$possible_registry" > /dev/null; then
+        repository="${registry_repository#*/}"
+        registry="$possible_registry"
+    else
+        registry="hub.docker.com"
+        repository="$registry_repository"
+    fi
+
+    if [ -z "$registry" ] || [ -z "$repository" ] || [ -z "$tag" ]; then
+        echo >&2 "'$1' is not a valid input." 
+        echo >&2 "Must include a repository and a tag."
+        return $ERROR_CODE_INVALID_INPUT
+    fi
+
+    original_repository="$repository"
+    if [ "$registry" = "hub.docker.com" ] && ! grep '/' <<< "$repository" > /dev/null; then
+        repository="library/$original_repository"
+    fi
+
+    t="$(mktemp)"
```

Comment:
Move this out + add an exit trap. No need to create + rm a ton of files, just do the one.